contracts/extensions/CoinMachine.sol

```diff
@@ -341,6 +359,15 @@ contract CoinMachine is ColonyExtension {
     );
   }
 
+  /// @notice Get the maximum amount of tokens a user can purchase
+  function getMaxPurchase(address _user) public view returns (uint256) {
```

Comment:
This function doesn't work at the start of a new period. Should this function also take in to account the remaining in the window, the token balance, etc? Basically fold in (most of) the `min` calls.

Reply:
My preference is to keep this context-free as much as possible -- returning the maximum the user can buy in general, and letting questions of period max, etc be handled elsewhere.

Reply:
Also, I don't follow that the function doesn't work at the start of a new period?

Reply:
> Also, I don't follow that the function doesn't work at the start of a new period?
Me neither. I'm going to blame the fumes I was casually inhaling all day.

That said, I do think there should be a separate function that does include all appropriate limits, or at the very least check with the frontend that they're happy doing all the required calculations, and the comment here should make it _very_ clear that the returned value might not, actually, be the amount of tokens you can purchase right now, which is a bit of a surprise for a public function called `getMaxPurchase`.

Reply:
Fair, maybe renaming the function `getUserLimit` would be better, and the variable `userLimitFraction`.